packages/block-library/src/button/editor.scss

```diff
@@ -70,52 +70,21 @@
 			text-overflow: ellipsis;
 		}
 	}
-}
-
-.block-library-button__inline-link {
-	background: $white;
-	display: flex;
-	flex-wrap: wrap;
-	align-items: center;
-	font-family: $default-font;
-	font-size: $default-font-size;
-	line-height: $default-line-height;
-
-	// The width of input box plus padding plus two icon buttons.
-	$blocks-button__link-input-width: 300px + 2px + 2 * $icon-button-size;
-	width: $blocks-button__link-input-width;
-
-	// Move it down by 2px so that it doesn't overlap the button focus outline.
-	margin-top: 2px;
-
-	.block-editor-url-input {
-		width: auto;
-	}
-
-	.block-editor-url-input__suggestions {
-		width: $blocks-button__link-input-width - $icon-button-size - $icon-button-size;
-		z-index: z-index(".block-library-button__inline-link .block-editor-url-input__suggestions");
-	}
-
-	> .dashicon {
-		width: $icon-button-size;
-	}
 
-	.dashicon {
-		color: $dark-gray-100;
-	}
-
-	.block-editor-url-input input[type="text"]::placeholder {
-		color: $dark-gray-100;
+	// Limit width of the text field if empty
+	.wp-block-button__link[data-is-placeholder-visible="true"] {
+		max-width: 150px;
 	}
+}
 
-	[data-align="center"] & {
-		margin-left: auto;
-		margin-right: auto;
-	}
+.wp-block-button__inline-link {
+	color: $dark-gray-500;
+	height: 0;
+	overflow: hidden;
 
-	[data-align="right"] & {
-		margin-left: auto;
-		margin-right: 0;
+	.is-selected & {
+		height: 100%;
```

Comment:
Minor, but I think `height: auto;` would be better, since `auto` is the default for `height` (https://developer.mozilla.org/en-US/docs/Web/CSS/height).

Reply:
Thank you! Addressed in [this commit](https://github.com/WordPress/gutenberg/pull/10128/commits/e2ed1dd10af1765de661a886996eccdccdbb9462) ❤️